### PR TITLE
MNTOR-4345: capture the edge case of orphan subscriber pref entries

### PR DESCRIPTION
**Pull request by mansaj**
# References:

Jira: MNTOR-4345

# Description
See incident channel

# How to test
* Have a pref table record for a subscriber
* Delete the subscriber record
* See the logs catching the error and updating the pref record

This simulates a subscriber in the same situation in prod

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/db/tables/subscriber_email_preferences.ts [hunk 1]
```diff
@@ -284,18 +285,50 @@ async function unsubscribeMonthlyMonitorReportForUnsubscribeToken(
...
+              monthly_monitor_report_free_at: knex.fn.now(),
+            })
+            .returning("*")
+        )?.[0];
+
+        logger.info(
+          "unsubscribe_monthly_monitor_report_for_unsubscribe_token_subscriber_not_found_updated",
+          {
+            subscriberId: sub.subscriber_id,
+            res,
```
[the remark this answers is not in the document]
Comment (mansaj) [reply src/db/tables/subscriber_email_preferences.ts:326]: i'll del that actually, subid is good enough
